tests/components/test-components.php

```diff
@@ -431,6 +442,167 @@ public function test_component_post_featured_media() {
 			],
 		];
 
-		$this->assertComponentEquals( $expected, 'irving/post-featured-media' );
+		$component = new Component( 'irving/post-featured-media' );
+
+		$this->assertComponentEquals( $expected, $component );
+	}
+
+	/**
+	 * Test irving/post-list component.
+	 *
+	 * @group core-components
+	 */
+	public function test_component_post_list() {
+		// Demo query args.
+		$query_args = [
+			'post__in' => [ $this->get_post_id() ],
+		];
+
+		// Demo templates.
+		$templates = [
+			'before'  => [
+				[ 'name' => 'example/before' ],
+			],
+			'after'   => [
+				[ 'name' => 'example/after' ],
+			],
+			'wrapper' => [ 'name' => 'example/wrapper' ],
+			'item'    => [ 'name' => 'example/item' ],
+		];
+
+		$expected = $this->get_expected_component(
+			'irving/post-list',
+			[
+				'_alias'   => 'irving/fragment',
+				'children' => [
+					$this->get_expected_component( 'example/before' ),
+					$this->get_expected_component(
+						'example/wrapper',
+						[
+							'children' => [
+								$this->get_expected_component(
+									'irving/post-provider',
+									[
+										'_alias'   => 'irving/fragment',
+										'config'   => [
+											'postId' => $this->get_post_id(),
+										],
+										'children' => [
+											$this->get_expected_component( 'example/item' ),
+										],
+									]
+								),
+							],
+						],
+					),
+					$this->get_expected_component( 'example/after' ),
+				],
+			]
+		);
+
+		$component = new Component(
+			'irving/post-list',
+			[
+				'config' => [
+					'query_args' => $query_args,
+					'templates'  => $templates,
+				],
+			]
+		);
+
+		$this->assertComponentEquals( $expected, $component );
+	}
+
+	/**
+	 * Test irving/post-list component without data.
+	 *
+	 * @group core-components
+	 */
+	public function test_component_post_list_no_results() {
+		$expected = $this->get_expected_component(
+			'irving/post-list',
+			[
+				'_alias'   => 'irving/fragment',
+				'children' => [
+					'No results found.',
+				],
+			]
+		);
+
+		$component = new Component(
+			'irving/post-list',
+			[
+				'children' => [ 'No results found.' ],
+			]
+		);
+
+		$this->assertComponentEquals( $expected, $component );
+	}
+
+	/**
+	 * Test irving/post-list component without data.
+	 *
+	 * @group core-components
+	 */
+	public function test_component_post_list_no_results_template() {
+		$expected = $this->get_expected_component(
+			'irving/post-list',
+			[
+				'_alias'   => 'irving/fragment',
+				'children' => [
+					$this->get_expected_component( 'example/none' ),
+				],
+			]
+		);
+
+		$component = new Component(
+			'irving/post-list',
+			[
+				'config'   => [
+					'templates' => [
+						'no_results' => [
+							[ 'name' => 'example/none' ],
+						],
+					],
+				],
+				'children' => [ 'No results found.' ],
+			]
+		);
+
+		$this->assertComponentEquals( $expected, $component );
+	}
+
+	/**
+	 * Helper for creating expected output for components.
```

Comment:
Got tired of writing the same defaults all the time. Maybe more confusing than it's worth?